### PR TITLE
feat: add cat vrs pydantic models

### LINKED ISSUE #2: create pydantic models
Create Pydantic models using the [1.x branch](https://github.com/ga4gh/cat-vrs/tree/1.x)

### PR DESCRIPTION
**Pull request by korikuzma**
close #2

### CODE REVIEW

**Comment by korikuzma** [comment]
Note for myself. This is up-to-date with current submodules. Just want to test in MetaKB first.

**Comment by korikuzma** [comment]
@DanielPuthawala it would be nice once things are more stable to create a test suite of valid and invalid catvars.

**Comment by DanielPuthawala** [comment]
Kori, are you intending to include the maturity status (draft, trial-use, normalized, deprecated) in these models?

File: src/ga4gh/cat_vrs/core_models.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+        :param v: copyChange value
+        :raises ValueError: If ``copyChange.code`` is not a valid CopyChange
+        :return: copyChange property
+        """
+        try:
+            CopyChange(v.code.root)
+        except ValueError as e:
+            err_msg = f"copyChange, {v.code.root}, not one of {[cc.value for cc in CopyChange]}"
+            raise ValueError(err_msg) from e
```
Comment (DanielPuthawala) [inline src/ga4gh/cat_vrs/core_models.py:75]: Do you represent the EFO codes enum anywhere?  I may be missing it, but I don't see it in here?

File: src/ga4gh/cat_vrs/profile_models.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    Constraint,
+    CopyChangeConstraint,
+    CopyCountConstraint,
+    DefiningContextConstraint,
+    Relation,
+)
+from pydantic import BaseModel, Field, field_validator
+
+
+class CatVrsType(str, Enum):
```
Comment (DanielPuthawala) [inline src/ga4gh/cat_vrs/profile_models.py:20]: So this is loss of a problem and more of a thin got keep in mind for later, there are an astronomically large number of possible CatVar types.  What we represent in `recipes` is only a very small subset of that, and do not constitute limits on valid CatVars, they just serve to define common useful CatVar classes of particular interest to the implementors.

**Comment by korikuzma** [comment]
> @korikuzma , are you intending to include the maturity status (draft, trial-use, normalized, deprecated) in these models?

@DanielPuthawala Since VRS-Python does not currently, I was not going to. However, in my Pydantic --> JSON Schema / RST I did include this. Can we do this in a separate issue?

File: src/ga4gh/cat_vrs/core_models.py (hunk 1 above)
Comment (korikuzma) [reply src/ga4gh/cat_vrs/core_models.py:75]: @DanielPuthawala these are imported from VRS-Python : https://github.com/ga4gh/vrs-python/blob/main/src/ga4gh/vrs/models.py#L171-L181

**Review by DanielPuthawala — APPROVED** [review]

**Pull request merged**